### PR TITLE
Feature/preview

### PR DESCRIPTION
**Pull request by akileng56**

### CODE REVIEW

File: src/ProgressCircle.xml [hunk 1]
```diff
@@ -19,6 +19,19 @@
             <category>Display</category>
             <description>Animates when the progress circle is loaded and updated</description>
         </property>
+        <property key="bootstrapStyle" type="enumeration" defaultValue="primary">
+            <caption>Bootstrap style</caption>
```
Comment (Andries-Smit) [inline src/ProgressCircle.xml]: No technical and implementation naming _Bootstrap style_ should be `Style`

File: src/components/ProgressCircle.ts [hunk 2]
```diff
@@ -4,21 +4,25 @@ import * as classNames from "classnames";
 import { Circle } from "progressbar.js";
 import { Alert } from "./Alert";
 
-import "../ui/ProgressCircle.css";
+import "../ui/ProgressCircle.scss";
 
-interface ProgressCircleProps {
+export interface ProgressCircleProps {
```
Comment (Andries-Smit) [inline src/components/ProgressCircle.ts:9]: Would be nice to sort it alphabetically, as your call are required to sorted by tslint.

File: src/components/ProgressCircle.ts [hunk 3]
```diff
@@ -48,14 +52,16 @@ class ProgressCircle extends Component<ProgressCircleProps, { alertMessage?: str
...
+            className: classNames("widget-progress-circle", this.props.className), style: this.props.style
+        },
             DOM.div({
-                className: classNames(`widget-progress-circle-${this.props.textSize}`,
+                className: classNames(
+                    `widget-progress-circle-${this.props.textSize} widget-progress-circle-${this.props.bootstrapStyle}`,
                     {
                         "widget-progress-circle-alert": typeof maximumValue === "number" ? maximumValue < 1 : false,
-                        "widget-progress-circle-clickable": this.props.clickable,
-                        "widget-progress-circle-negative": !!value && value < 0
```
Comment (Andries-Smit) [inline src/components/ProgressCircle.ts]: Why remove this?

File: src/components/ProgressCircleContainer.ts [hunk 4]
```diff
@@ -1,10 +1,16 @@
 import { Component, createElement } from "react";
-import { ProgressCircle, ProgressTextSize } from "./ProgressCircle";
+import { BootstrapStyle, ProgressCircle, ProgressTextSize } from "./ProgressCircle";
 import { Alert } from "./Alert";
 
-interface ProgressCircleContainerProps {
-    animate: boolean;
+interface WrapperProps {
+    class?: string;
```
Comment (Andries-Smit) [inline src/components/ProgressCircleContainer.ts]: Wrapper props class and style are not optional but `mxObject `is style and class are empty string when they contain no value

File: typings/progressCircle.d.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+declare module "*.scss";
```
Comment (Andries-Smit) [inline typings/progressCircle.d.ts]: missing new line

File: typings/customMatchers.d.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
-// Note: cannot use ReactElement here, because importing a module from the declaration of a namespace is not allowed.
-declare namespace jasmine {
-    interface Matchers<T> {
-        toBeElement(expected: any): boolean;
-        toHaveClass(expected: string): boolean;
-        toMatchStructure(expected: any): boolean;
-    }
-}
+// Note: cannot use ReactElement here, because importing a module from the declaration of a namespace is not allowed.
```
Comment (Andries-Smit) [inline typings/customMatchers.d.ts:1]: Nothing changes, what is the LF or CRLF ?

File: src/components/ProgressCircle.ts [hunk 7]
```diff
@@ -48,14 +52,16 @@ class ProgressCircle extends Component<ProgressCircleProps, { alertMessage?: str
...
-        const { maximumValue, value } = this.props;
-        return DOM.div({ className: "widget-progress-circle" },
+        const { maximumValue } = this.props;
+        return DOM.div({
+            className: classNames("widget-progress-circle", this.props.className), style: this.props.style
+        },
             DOM.div({
-                className: classNames(`widget-progress-circle-${this.props.textSize}`,
+                className: classNames(
+                    `widget-progress-circle-${this.props.textSize} widget-progress-circle-${this.props.bootstrapStyle}`,
```
Comment (Andries-Smit) [inline src/components/ProgressCircle.ts]: 2 class names, should be separated by coma.
```
`widget-progress-circle-${this.props.textSize},
widget-progress-circle-${this.props.bootstrapStyle}`,
```

File: src/components/__tests__/ProgressCircle.spec.ts [hunk 8]
```diff
@@ -119,12 +120,6 @@ describe("ProgressCircle", () => {
         expect(progressCircle.animate).toHaveBeenCalledWith(1);
     });
 
-    it("has the class widget-progress-circle-negative when the value is less than zero", () => {
```
Comment (Andries-Smit) [inline src/components/__tests__/ProgressCircle.spec.ts:122]: You just can not remove a feature. without consulting the client. This is a breaking change. May be the boostrap style for negative value should be configured too, please check.

File: tslint.json [hunk 9]
```diff
@@ -84,7 +84,8 @@
             }
         ],
         "curly": false,
-        "arrow-parens": false
+        "arrow-parens": false,
+        "linebreak-style": [ true, "LF" ]
     },
     "jsRules": {
         "trailing-comma": [ false ],
```
Comment (Andries-Smit) [inline tslint.json:91]: For js file also add 
`"linebreak-style": [ true, "LF" ]`

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/ProgressCircle.xml (hunk 1 above)
Comment (akileng56) [reply src/ProgressCircle.xml]: Fixed

File: src/components/ProgressCircle.ts (hunk 2 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts:9]: Done

File: src/components/ProgressCircle.ts (hunk 7 above)
Comment (akileng56) [reply src/components/ProgressCircle.ts]: Done

File: src/components/__tests__/ProgressCircle.spec.ts (hunk 8 above)
Comment (akileng56) [reply src/components/__tests__/ProgressCircle.spec.ts:122]: Will confirm this with Danny

File: src/components/ProgressCircleContainer.ts (hunk 4 above)
Comment (akileng56) [reply src/components/ProgressCircleContainer.ts]: Fixed

File: typings/progressCircle.d.ts (hunk 5 above)
Comment (akileng56) [reply typings/progressCircle.d.ts]: Added

File: tslint.json (hunk 9 above)
Comment (akileng56) [reply tslint.json:91]: Added

**Pull request merged**